includes/admin/tools/class-edd-tools-recount-customer-stats.php

```diff
@@ -66,40 +66,22 @@ public function get_data() {
 
 			foreach ( $customers as $customer ) {
 
-				$attached_payment_ids = explode( ',', $customer->payment_ids );
-
-				$attached_args = array(
-					'post__in' => $attached_payment_ids,
-					'number'   => -1,
-					'status'   => $allowed_payment_status,
-				);
-
-				$attached_payments = edd_get_payments( $attached_args );
-
-				$unattached_args = array(
-					'post__not_in' => $attached_payment_ids,
-					'number'       => -1,
-					'status'       => $allowed_payment_status,
-					'meta_query'   => array(
-						array(
-							'key'     => '_edd_payment_user_email',
-							'value'   => $customer->email,
-							'compare' => '=',
-						)
-					),
-				);
-
-				$unattached_payments = edd_get_payments( $unattached_args );
-
-				$payments = array_merge( $attached_payments, $unattached_payments );
+				$payment_ids = edd_get_orders( array(
+					'type'        => 'sale',
+					'customer_id' => $customer->id,
+					'status__in'  => $allowed_payment_status,
+					'number'      => 999,
```

Comment:
Is 999 a large enough number?

Reply:
I think so. This would be orders for a single customer in the allowed status list, that are sales only. Based off those numbers the customer would be able to have the following to burst past 999:
2.7 years of daily purchases or
19.21 years of 1 purchase a week or
83 years of a customer making a single purchase per month.

Based off those numbers, I think it's pretty safe to assume that 999 is within reason.